src/d3a/models/myco_matcher/external_matcher.py

```diff
@@ -1,140 +1,179 @@
+from d3a.d3a_core.exceptions import (
+    InvalidBidOfferPairException, OfferNotFoundException,
+    BidNotFoundException, MycoValidationException)
 from d3a.d3a_core.redis_connections.redis_area_market_communicator import ResettableCommunicator
-from d3a.models.market import validate_authentic_bid_offer_pair
-from d3a.models.market.market_structures import (
-    BidOfferMatch, offer_or_bid_from_json_string)
+from d3a.models.market import Market
+from d3a.models.market.market_structures import BidOfferMatch
 
 from d3a.models.myco_matcher.base_matcher import BaseMatcher
 
 
 class ExternalMatcher(BaseMatcher):
+    """Class responsible for external bids / offers matching."""
     def __init__(self):
         super().__init__()
         self.simulation_id = d3a.constants.COLLABORATION_ID
         self.myco_ext_conn = None
-        self.channel_prefix = f"external-myco/{self.simulation_id}/"
+        self.channel_prefix = f"external-myco/{self.simulation_id}"
+        self.response_channel = f"{self.channel_prefix}/response"
+        self.events_channel = f"{self.response_channel}/events/"
         self._setup_redis_connection()
         self.area_uuid_markets_mapping = {}
-        self.markets_mapping = {}
+        self.markets_mapping = {}  # Dict[market_id: market] mapping
         self.recommendations = []
 
     def _setup_redis_connection(self):
         self.myco_ext_conn = ResettableCommunicator()
         self.myco_ext_conn.sub_to_multiple_channels(
-            {"external-myco/get-simulation-id": self.get_simulation_id,
-             f"{self.channel_prefix}offers-bids/": self.publish_offers_bids,
-             f"{self.channel_prefix}post-recommendations/": self.match_recommendations})
+            {"external-myco/get-simulation-id": self.publish_simulation_id,
+             f"{self.channel_prefix}/offers-bids/": self.publish_offers_bids,
+             f"{self.channel_prefix}/post-recommendations/": self.match_recommendations})
 
     def publish_offers_bids(self, message):
         """Publish open offers and bids.
 
-        published data are of the following format
-        market_offers_bids_list_mapping = {"market_id" : {"bids": [], "offers": [] }, }
+        published data are of the following format:
+            {"bids_offers": {`market_id` : {"bids": [], "offers": [] }, }}
         """
-        # TODO: message can contain filters
-        data = {"event": "offers_bids_response"}
+        response_data = {"event": "offers_bids_response"}
+        data = json.loads(message.get("data"))
+        filters = data.get("filters", {})
+        # IDs of markets (Areas) the client is interested in
+        filtered_market_ids = filters.get("markets", None)
         market_offers_bids_list_mapping = {}
-        for area_uuid, markets in self.area_uuid_markets_mapping.items():
+        for area_id, markets in self.area_uuid_markets_mapping.items():
+            if filtered_market_ids and area_id not in filtered_market_ids:
```

Comment:
Regarding naming: reading `area_id not in filtered_market_ids` feels strange to me. I would expect to find market IDs in the `filtered_market_ids` variable.
Maybe we could use a name like `areas_to_market_ids`? 

On a different note: why are we using `_id` and not `_uuid`?

Reply:
I absolutely have no idea why I renamed it to area_id, I faintly recall this was discussed somewhere but I'm not sure
I'll fix this